clients/cpp/src/signalrlib/HubConnection.cpp

```diff
@@ -0,0 +1,201 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+#include "HubConnection.h"
+#include "WebSocketsTransport.h"
+
+namespace signalr
+{
+    enum MessageType
+    {
+        Invocation = 1,
+        StreamItem,
+        Completion,
+        StreamInvocation,
+        CancelInvocation,
+        Ping,
+        Close,
+    };
+
+    HubConnection::HubConnection(const utility::string_t& url, Transport transport)
+        : mHandshakeReceived(false)
+    {
+        mUrl = url;
+        mTransport = new WebSocketsTransport(mUrl);
+        mTransport->OnReceived([&](const utility::string_t& messages)
```

Comment:
don't capture like this - this will crash very hard. I did not read the code very carefully but I think this lambda can outlive the connection and all your m* variables will cause segfaults. There is a reason while the old SignalR C++ client used shared_ptr all over the place (https://github.com/SignalR/SignalR-Client-Cpp/blob/dev/src/signalrclient/connection_impl.cpp#L96). Read more on this in my post https://blog.3d-logic.com/2014/12/28/c-async-development-not-only-for-for-c-developers-part-ii-lambda-functions-the-most-common-pitfalls/
(because you are using Casablanca consider reading entire series: https://blog.3d-logic.com/category/cpprestsdk/ - it is basically my learnings from writing the SignalR client. Sadly, I have never written the last two posts).

Reply:
Yeah, I'm going to be using shared_ptr later, this was a quick, get things working